server/channels/jobs/batch_worker.go

```diff
@@ -18,10 +18,11 @@ type BatchWorker struct {
 	logger    mlog.LoggerIFace
 	store     store.Store
 
-	stop    chan struct{}
-	stopped chan bool
-	closed  atomic.Bool
-	jobs    chan model.Job
+	stateMut sync.Mutex
```

Comment:
Can we add a comment to make clear what is protected by the mutex? I was expecting, it's the whole block, but e.g. `jobs` isn't.